server/src/internalClusterTest/java/org/opensearch/index/codec/CodecCompressionLevelIT.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec;
+
+import org.apache.logging.log4j.core.util.Throwables;
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
+public class CodecCompressionLevelIT extends OpenSearchIntegTestCase {
```

Comment:
I'm wondering if a lighter weight unit test would be more appropriate here? Like possibly in `org.opensearch.index.codec.CodecTests`? It looks like you're just testing setting validations, which seems like a unit test should be able to do and not require a full integration test setup.

Reply:
I was testing out scenarios of updating index settings before/after the index has been created. If it makes sense to only have UTs in place, then I can refactor